File: bindings/ruby/testsuite/logger.rb
#!/usr/bin/ruby

require 'test/unit'
require 'storage'


$my_logger_called = false


class MyLogger < Storage::Logger

  def write(level, component, filename, line, function, content)

    puts "my-logger " + content

    $my_logger_called = true

  end

end


class TestLogger < Test::Unit::TestCase

  def test_logger

    @my_logger = MyLogger.new
    Storage::logger = @my_logger

    GC.start

    # Just test that our logger object is called and not already garbage
    # collected.

    environment = Storage::Environment.new(true, Storage::ProbeMode_NONE, Storage::TargetMode_DIRECT)
    storage = Storage::Storage.new(environment)

    assert($my_logger_called)

  end

end

File: bindings/python/testsuite/logger.py
#!/usr/bin/python3

import unittest, gc
from storage import Logger, set_logger, Environment, ProbeMode_NONE, TargetMode_DIRECT, Storage


my_logger_called = False


class MyLogger(Logger):

    def write(self, level, component, filename, line, function, content):

        print("my-logger", content)

        global my_logger_called
        my_logger_called = True


class TestLogger(unittest.TestCase):

    def test_logger(self):

        my_logger = MyLogger()
        set_logger(my_logger)

        gc.collect()

        # Just test that our logger object is called and not already garbage
        # collected.

        environment = Environment(True, ProbeMode_NONE, TargetMode_DIRECT)
        s = Storage(environment)

        self.assertTrue(my_logger_called)


if __name__ == '__main__':
    unittest.main()
